src/Ocelot/Configuration/File/FileAuthenticationOptions.cs

```diff
@@ -17,11 +17,22 @@ public FileAuthenticationOptions(FileAuthenticationOptions from)
 
         public List<string> AllowedScopes { get; set; }
 
+        /// <summary>
+        /// Allows anonymous authentication for route when global AuthenticationOptions are used.
+        /// </summary>
+        /// <value>
+        /// <see langword="true"/> if it is allowed; otherwise, <see langword="false"/>.
+        /// </value>
+        public bool AllowAnonymous { get; set; }
+
         [Obsolete("Use the " + nameof(AuthenticationProviderKeys) + " property!")]
         public string AuthenticationProviderKey { get; set; }
 
         public string[] AuthenticationProviderKeys { get; set; }
 
+        public bool HasProviderKey() => !string.IsNullOrEmpty(AuthenticationProviderKey)
```

Comment:
Single-line expression should be a property rather than a method.
```suggestion
        public bool HasProviderKey => !string.IsNullOrEmpty(AuthenticationProviderKey)
```